fl4health/server/base_server.py

```diff
@@ -369,10 +362,32 @@ def _terminate_after_unacceptable_failures(self, timeout: Optional[float]) -> No
         # First we shutdown all clients involved in the FL training/evaluation if they can be.
         self.disconnect_all_clients(timeout=timeout)
         # Throw an exception alerting the user to failures on the client-side causing termination
+        self.shutdown()
         raise ValueError(
             f"The server encountered failures from the clients and accept_failures is set to {self.accept_failures}"
         )
 
+    def _log_client_failures(self, failures: FitFailures | EvaluateFailures) -> None:
+        log(
+            ERROR,
+            f"There were {len(failures)} failures in the fitting process. This will result in termination of "
+            "the FL process",
+        )
+        for failure in failures:
+            if isinstance(failure, BaseException):
+                log(
+                    ERROR,
+                    "An exception was returned instead of any failed results. As such the client ID is unknown. "
```

Comment:
This might be a bit overkill so feel free to ignore this suggestion. But could you not technically determine the client ids of the clients that returned exceptions by process of elimination? If you have multiple exceptions you won't know which exception goes with which client, but you could at least log a list of clients that failed.

Reply:
Yup. That's a fair point. I'll log a list of clients that failed. We won't know which corresponds to which exception, but it's reasonable to at least gather them.

Reply:
So it turns out this is harder than it seems at first. Basically, if they return only an exception, they are immediately removed from the client manager and they aren't necessarily connected to the server yet when the first fit_round starts (for example). So the representation of these CIDs in the client manager is a bit asynchronous, which makes these reporting really difficult. I'm going to just punt on it for now and we can try to be more verbose later.